src/context/prop.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {devAssert} from '../log';
+
+const EMPTY_DEPS = [];
+
+/**
+ * A context property.
+ *
+ * @interface
+ * @template T
+ */
+export class ContextProp {
+  /**
+   * A globally unique key. Extensions must use a fully qualified name such
+   * as "amp-extension:key" or "amp-extension:version:key".
+   *
+   * @return {string}
+   */
+  get key() {}
+
+  /**
+   * An optional type object that can be used for a using system. E.g.
+   * this could be a Preact's Context object.
+   *
+   * @return {?Object}
+   */
+  get type() {}
+
+  /**
+   * An array of dependencies that are required for the `compute` callback.
+   *
+   * @return {?Array<!ContextProp>}
```

Comment:
Why is this `?Array`? Looks like it is initialized to `[]` and treated as if it will have an integer `length`

Reply:
Fixed.